filters/auth/auth.go

```diff
@@ -31,3 +60,21 @@ func (authFilter AuthenticationFilter) basicAuth(req *web.Request, handler web.H
 
 	return handler(req)
 }
+
+// Filter which authenticates requests coming from Service Manger CLI using OAuth
+func (authFilter AuthenticationFilter) oAuth(req *web.Request, handler web.Handler) (*web.Response, error) {
+	authenticator, err := oidc.NewAuthenticator(req.Request.Context(), oidc.Options{
+		IssuerURL: authFilter.TokenIssuerURL,
+		ClientID: "cf",
+	})
+	if err != nil {
+		return nil, err
```

Comment:
I think it might be better to wrap the err in `web.NewHTTPError`, otherwise it will return 500 Internal Server Error, which is not always the case here.